createtemplates.sh

```diff
@@ -0,0 +1,327 @@
+#!/bin/bash
+
+# console url
+BASE_URL="https://console.stage.rafay.dev"
+API_KEY="ra2.aca8acd81c49fa393364ab744574a6d5386ec7f6.46166dc1e95bc8c12670999c93d02d4285bd02b33f1a317a8d737701810dd7b7"
+
+PROJECT_NAME="templates"
+PROJECT_HASH=""
+REPO_NAME="rafay-repository"
+PROJECT_NAME_FIELD="projectName"
+
+GET_PROJECTS="v1/auth/projects"
+ADD_RESOURCE_TEMPLATE="apis/eaas.envmgmt.io/v1/projects/${PROJECT_NAME}/resourcetemplates"
+ADD_ENVIRONMENT_TEMPLATE="apis/eaas.envmgmt.io/v1/projects/${PROJECT_NAME}/environmenttemplates"
+ADD_DRIVER_TEMPLATE="apis/eaas.envmgmt.io/v1/projects/${PROJECT_NAME}/drivers"
+ADD_CONFIGCONTEXT_TEMPLATE="apis/eaas.envmgmt.io/v1/projects/${PROJECT_NAME}/configcontexts"
+CREATE_PROJECT_URL="auth/v1/projects"
+
+CLEANUP_TEMP_FILES=true
+
+
+# list of all templates which needs to be created
+templates=(
+    "terraform/naas/101-naas"
+    #"terraform/aws/101-caas-eks"
+    #"terraform/aws/101-caas-ecs"
+    #"terraform/aws/101-genai-ecs"
+    #"terraform/aws/101-genai-eks"
+    #"terraform/aws/101-kubeflow-eks"
+    #"terraform/aws/101-naas-eks"
+    #"terraform/aws/101-vpc-ec2"
+    #"terraform/aws/101-vpc-ecs"
+    #"terraform/aws/101-waas-eks"
+    #"terraform/azure/101-vnet-instance"
+    #"terraform/gcp/101-caas-gke"
+    #"terraform/gcp/101-vpc-instance"
+    #"terraform/mks/101-caas-pnap"
+    #"terraform/oci/101-caas-oke"
+    #"terraform/vcluster/101-caas"
+    #"terraform/vmware/101-caas-vmware"
+    #"terraform/waas/101-waas"
+)
+
+# POST request for HUB apis
+function make_get_request_new {
+    local endpoint="$1"
+    curl -s -X GET -H "Accept */*" -H "X-API-KEY: ${API_KEY}" "${BASE_URL}/${endpoint}"
```

Comment:
Should we do some error handling for all the API requests to make it easier to debug in case of failure?